llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -8741,6 +8744,12 @@ Value *SPIRVToLLVM::transGLSLExtInst(SPIRVExtInst *extInst, BasicBlock *bb) {
 
   case GLSLstd450Fma:
     // Fused multiply and add
+    if (m_shaderOptions->disableFMA || DisableFMA) {
+      auto fmf = getBuilder()->getFastMathFlags();
+      fmf.setAllowContract(false);
+      getBuilder()->setFastMathFlags(fmf);
+      return getBuilder()->CreateFAdd(getBuilder()->CreateFMul(args[0], args[1]), args[2]);
+    }
```

Comment:
I believe there should be an `IRBuilderBase::FastMathFlagGuard` here.

Reply:
Fixed. Thanks.